lib/ethereum/contract.rb

```diff
@@ -166,12 +166,20 @@ def call_payload(fun, args)
       "0x" + fun.signature + (@encoder.encode_arguments(fun.inputs, args).presence || "0"*64)
     end
 
-    def call_args(fun, args)
-      add_gas_options_args({to: @address, from: @sender, data: call_payload(fun, args)})
+    def call_args(tx_args, fun, args)
+      argsHash = {}
+      argsHash[:to] = @address
+      argsHash[:from] = @sender
+      argsHash[:data] = call_payload(fun, args)
+      argsHash[:value] = "0x#{tx_args[:value]}" if !tx_args.nil? && tx_args.key?(:value) && !tx_args[:value].nil?
```

Comment:
Simpler:
`argsHash[:value] = "0x#{tx_args[:value]}" if tx_args && tx_args[:value]`

Reply:
Note: this assumes we provide value in hex, I would assume we provide value as a normal integer.